app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt

```diff
@@ -131,27 +137,89 @@ class HtmlParserTest {
 
   @Test
   fun testHtmlContent_customSpan_isAdded() {
-    val textView = activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
-    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */ "", /* imageCenterAlign= */ true)
+    val textView =
+      activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlParser = htmlParserFactory.create(
+      /* entityType= */ "",
+      /* entityId= */ "",
+      /* imageCenterAlign= */true
+    )
     val htmlResult: Spannable = htmlParser.parseOppiaHtml(
       "<p>You should know the following before going on:<br></p>" +
-          "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
-          "<li>How to tell whether one counting number is bigger or smaller than another<br></li></ul>",
+        "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
+        "<li>How to tell whether one counting number is bigger or smaller than another<br></li></ul>",
       textView
     )
 
     /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
-    val bulletSpans = htmlResult.getSpans<CustomBulletSpan>(0, htmlResult.length, CustomBulletSpan::class.java)
+    val bulletSpans =
+      htmlResult.getSpans<TextLeadingMarginSpan>(0, htmlResult.length, TextLeadingMarginSpan::class.java)
     assertThat(bulletSpans.size.toLong()).isEqualTo(2)
 
-    val bulletSpan0 = bulletSpans[0] as CustomBulletSpan
+    val bulletSpan0 = bulletSpans[0] as TextLeadingMarginSpan
     assertThat(bulletSpan0).isNotNull()
 
-    val bulletSpan1 = bulletSpans[1] as CustomBulletSpan
+    val bulletSpan1 = bulletSpans[1] as TextLeadingMarginSpan
     assertThat(bulletSpan1).isNotNull()
   }
 
-  @Qualifier annotation class TestDispatcher
+  @Test
+  fun testHtmlContent_textLeadingMarginSpan_isAdded() {
+    val textView =
+      activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlParser = htmlParserFactory.create(
+      /* entityType= */ "",
+      /* entityId= */ "",
+      /* imageCenterAlign= */ true
+    )
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+      """
+           <ul>
+  <li>Item 1</li>
+  <li>Item 2</li>
+  <li>Numbered list:
+    <ol>
+      <li>Nested item in numbered list 1</li>
+      <li>Nested item in numbered list 2</li>
+    </ol>
+  </li>
+  <li>Nested list:
+    <ul>
+      <li>Double nested list:
+        <ul>
+          <li>
+            Double nested item
+          </li>
+        </ul>
+      </li>
+    </ul>
+  </li>
+</ul>
+        """,
+      textView
+    )
+
+// get all the spans attached to the SpannedString
```

Comment:
Probably don't need this comment since the code seems pretty clear.

Reply:
Removed this testcase.